pkg/apis/agones/v1/gameserver.go

```diff
@@ -926,3 +926,95 @@ func (gs *GameServer) Patch(delta *GameServer) ([]byte, error) {
 	result, err = json.Marshal(patch)
 	return result, errors.Wrapf(err, "error creating json for patch for GameServer %s", gs.ObjectMeta.Name)
 }
+
+// UpdateCount increments or decrements a CounterStatus on a Game Server by the given amount.
+func (gs *GameServer) UpdateCount(name string, action string, amount int64) error {
+	err := fmt.Errorf("unable to UpdateCount: Name %s, Action %s, Amount %d", name, action, amount)
```

Comment:
Same as below, I'm wondering if a specific error for each condition would be easier to understand if they bubble up to the user.

Also - just realised too - please use the `errors` package, just because consistency (and we can get stack traces from them)